### ISSUE TITLE
No Documention

### ISSUE THREAD

**Issue by ivanjeremic** [atlassian/react-beautiful-dnd#1933]
How about adding documentation for react-beautiful-dnd? Storybook is not documentation and it has 0 lines of code examples.

**Comment by imjoshellis** [comment]
I agree. I actually just released a tutorial on basic minimal multi-column drag and drop. [Repo](https://github.com/imjoshellis/react-beautiful-dnd-tutorial) & [Blog](https://dev.to/imjoshellis/codealong-multi-column-drag-and-drop-in-react-3781)

I'd be happy to reformat and reorganize it as needed and submit a PR here if the maintainer chimes in.

**Comment by willmanduffy** [comment]
There's a fair bit of documentation already in the README. You can also browse the files here: https://github.com/atlassian/react-beautiful-dnd/tree/master/docs

There's also a full getting started video course linked from the README: https://egghead.io/courses/beautiful-and-accessible-drag-and-drop-with-react-beautiful-dnd

As a developer I've found these largely sufficient for all my needs!

**Comment by hoangvvo** [comment]
I recommend drop https://github.com/docsifyjs/docsify/ in. This creates a doc page with a sidebar with headers auto populated based on headings in the markdown files.

**Comment by ivanjeremic** [comment]
> There's a fair bit of documentation already in the README. You can also browse the files here: https://github.com/atlassian/react-beautiful-dnd/tree/master/docs
> 
> There's also a full getting started video course linked from the README: https://egghead.io/courses/beautiful-and-accessible-drag-and-drop-with-react-beautiful-dnd
> 
> As a developer I've personally found these largely sufficient for all my needs!

I don't think this is enough for a library that popular. People want real docs where they can copy code from the examples and paste it in their projects. Digging the code samples out of the GitHub repo is not a solution, finding the storybook and the components, then find out where the import is coming from is not how we should do it in 2020.

**Comment by willmanduffy** [comment]
I personally do not consider it "digging in the repo" if you can find it all in the README. https://github.com/atlassian/react-beautiful-dnd#documentation- Using GitHub README's as documentation, while not my personal favorite approach to docs, is perfectly valid and in wide use in libraries. There are plenty of code samples to copy and paste from there and if you need more, Storybook is code and is in the codebase.

![2020-09-08 08 21 02](https://user-images.githubusercontent.com/782252/92475999-486d1a00-f1ac-11ea-9522-f3664c8ef69a.gif)

You are right though! There's a lot of different formats for documentation that might be a better fit! Would love to see what you can come up with.

**Comment by ivanjeremic** [comment]
> I personally do not consider it "digging in the repo" if you can find it all in the README. https://github.com/atlassian/react-beautiful-dnd#documentation- Using GitHub README's as documentation, while not my personal favorite approach to docs, is perfectly valid and in wide use in libraries. There are plenty of code samples to copy and paste from there and if you need more, Storybook is code and is in the codebase.
> 
> ![2020-09-08 08 21 02](https://user-images.githubusercontent.com/782252/92475999-486d1a00-f1ac-11ea-9522-f3664c8ef69a.gif)
> 
> You are right though! There's a lot of different formats for documentation that might be a better fit! Would love to see what you can come up with.

I disagree that it is enaugh

**Comment by willmanduffy** [comment]
Super excited to see what you come up with!

**Comment by robhanlon22** [comment]
I am for this—I would love a more standard set of API docs that aren't guides would be really helpful, especially because to really grok how everything works together, ones needs to have many of the guides open at once and cross-reference. Having a direct component reference with all of the inputs and outputs would make it easier to jump in and experiment.
